templates/zerver/api/construct-narrow.md

```diff
@@ -0,0 +1,86 @@
+# Construct a narrow
+
+A **narrow** is a set of filters for Zulip messages, that can be based on many
+different factors (like sender, date, stream...).
+
+Narrows look like this:
+
+```
+stream:announce sender:iago@zulip.com
+```
+
+## Operators and operands
+
+Each of the key-value pairs that make a narrow are a filter for a specific
+characteristic of the messages. Here are the message filters (or
+**operators**) you can use for a narrow:
+
+* `search`: contain specific keywords. Example: `search:cake`.
+* `stream`: sent to a specific stream. Example: `stream:sales`.
+* `topic`: sent to a specific topic. Example: `topic:party`.
+* `sender`: sent by a specific user.
+    Example: `sender:aaron@zulip.com`.
+* `is`: belong to one of the following categories.
+    Example: `is:private`.
+    * `private`: sent to this user as a PM.
+    * `starred`: starred by the user making the narrow.
+    * `unread`: have not been read by the user making the narrow.
+    * `alerted`: contain one or more of the alert words configured by
+        the user making the narrow.
+    * `mentioned`: contain an `@mention` for the user making the narrow.
+* `has`: contain one or more of the following. Example: `has:link`.
+    * `link`
+    * `image`
+    * `attachment`
+* `near`: sent around some other. The other message has to be specified with
+    its ID. Example: `near:314`.
+
+The values that each operator carry are the **operands**. Thus, in the
+following narrow:
+
+```
+stream:holidays search:where+are+my+sunglasses?
+```
+
+<table>
+    <tr>
+        <th>Operator</th>
+        <th>Operand</th>
+    </tr>
+    <tr>
+        <td>stream</td>
+        <td>holidays</td>
+    </tr>
+    <tr>
+        <td>search</td>
+        <td>where are my sunglasses?</td>
+    </tr>
+</table>
+
+Note that you can escape the spaces with the plus (`+`) character when you're
+using the `key:value` narrow syntax.
+
+## JSON-based narrows
+
+While `key:value` can be used in Zulip's search bar, the narrows used by
+endpoints in our REST API are structured in JSON format.
+
+The result is an array of objects, each with an `operator` and an `operand`
+key.
+
+This way, the previous example would look like this:
+
+```
+[
+    {
+        "operator": "stream",
+        "operand": "holidays"
+    },
+    {
+        "operator": "search",
+        "operand": "where are my sunglasses?"
+    }
+]
+```
+
+Note that escaping spaces isn't required here.
```

Comment:
This should probably be just an issue for future work, but it'd be super cool to have a tool that converts a URL for a narrow into the corresponding JSON data format.

Reply:
That could be something fun to work on. Any ideas on how could we embed such a widget in these docs? Adding the HTML, JS, etc. right here is definitely going to be very messy.